src/frequenz/dispatch/_dispatcher.py

```diff
@@ -214,15 +216,29 @@ def __init__(
         Args:
             microgrid_id: The microgrid id.
             server_url: The URL of the dispatch service.
-            key: The key to access the service.
+            key: The key to access the service, deprecated, use `auth_key` instead.
+            auth_key: The authentication key to access the service.
+            sign_secret: The secret to sign the requests, optional
             call_timeout: The timeout for API calls.
             stream_timeout: The timeout for streaming API calls.
+
+        Raises:
+            ValueError: If both `key` and `auth_key` are provided.
         """
         super().__init__()
 
+        if key is not None and auth_key is not None:
+            raise ValueError(
+                "Both 'key' and 'auth_key' are provided, please use only 'auth_key'."
+            )
+
+        if key is not None:
+            auth_key = key
```

Comment:
It would be silly to do, but maybe also add an exception for `if not key and not auth_key`? Or will this be caught by `DispatcherApiClient()`?